can/io/trc.py

```diff
@@ -0,0 +1,364 @@
+# coding: utf-8
+
+"""
+Reader and writer for can logging files in peak trc format
+
+See https://www.peak-system.com/produktcd/Pdf/English/PEAK_CAN_TRC_File_Format.pdf
+for file format description
+
+Version 1.1 will be implemented as it is most commonly used
+"""  # noqa
+
+from typing import cast, Any, Generator, IO, Optional
+from datetime import datetime, timedelta
+from enum import Enum
+from io import TextIOWrapper
+import os
+import logging
+
+from ..message import Message
+from ..listener import Listener
+from ..util import channel2int
+from .generic import BaseIOHandler
+
+
+logger = logging.getLogger("can.io.trc")
+
+
+class TRCFileVersion(Enum):
+    UNKNOWN = 0
+    V1_0 = 100
+    V1_1 = 101
+    V1_2 = 102
+    V1_3 = 103
+    V2_0 = 200
+    V2_1 = 201
+
+
+class TRCReader(BaseIOHandler):
+    """
+    Iterator of CAN messages from a TRC logging file.
+    """
+
```

Comment:
```suggestion

    file: TextIO

```
This should help with mypy. You need to import TextIO of course.